pallets/gear/src/runtime_api.rs

```diff
@@ -40,11 +42,204 @@ impl<T: Config> Pallet<T>
+        let value_for_gas = <T as pallet_gear_bank::Config>::GasMultiplier::get().gas_to_value(gas);
+
+        // Required balance of the account.
+        let required_balance = CurrencyOf::<T>::minimum_balance()
+            .saturating_add(value_for_gas)
+            .saturating_add(value);
+
+        // Updating balance of the account.
+        let _ = CurrencyOf::<T>::deposit_creating(
+            &origin,
+            required_balance.saturating_sub(origin_balance),
+        );
+
+        // Returning origin account as signed origin.
+        RawOrigin::Signed(origin)
+    }
+
+    // Converts given dispatch error into dedicated runtime api string format.
+    fn dispatch_err_to_string(
+        extrinsic_name: &'static str,
+        e: DispatchErrorWithPostInfo<PostDispatchInfo>,
+    ) -> String {
+        // Extracting index of module returned error, if possible.
+        let error_module_idx = match e.error {
+            DispatchError::Module(ModuleError { index, .. }) => Some(index as usize),
+            _ => None,
+        };
+
+        // Converting dispatch error into string representation in default impl.
+        let error_message: &'static str = e.into();
+
+        // Creating result message.
+        let mut res = format!("Extrinsic `gear.{extrinsic_name}` failed: '{error_message}'");
+
+        // Extracting `pallet_gear` index from runtime to compare with dispatch error.
+        let Some(gear_module_idx) = PalletInfoOf::<T>::index::<Self>() else {
+            return Self::internal_err_string("No index found for `pallet_gear` in the runtime");
+        };
+
+        // Appending result message with pallet index returned error, if not this.
+        if let Some(module_idx) = error_module_idx.filter(|i| *i != gear_module_idx) {
+            res = format!("{res} (pallet index of the error: {module_idx}");
+        }
+
+        res
+    }
+
+    // Queries first element of the queue and extracts its message id.
+    fn queue_head() -> Result<MessageId, String> {
+        QueueOf::<T>::iter()
+            .next()
+            .ok_or_else(|| Self::internal_err_string("Failed to get last message from the queue"))
+            .and_then(|queued| {
+                queued
+                    .map(|dispatch| dispatch.id())
+                    .map_err(|_| Self::internal_err_string("Failed to extract queued dispatch"))
+            })
+    }
+
+    // Formats given message into dedicated runtime api string format.
+    fn internal_err_string(message: impl ToString) -> String {
+        format!(
+            "Internal error: entered unreachable code '{}'",
+            message.to_string()
+        )
+    }
+
+    // Returns none if queue is empty, otherwise - associated journal and bool,
+    // defining was it processed by builtin actor or not.
+    fn dequeue_head_and_run(
+        ext_manager: &mut ExtManager<T>,
+        builtin_dispatcher: &impl BuiltinDispatcher,
+    ) -> Result<Option<(Vec<JournalNote>, bool)>, String> {
```

Comment:
What the reason to return this bool ?

Reply:
I'll reuse this code in calculate gas impl, because its 99% similar, but calculate gas needs to know who was processing it